src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -101,6 +123,88 @@ public void resetOnPath(String path) {
         // Intentionally blank
     }
 
+    /**
+     * Helper method which handles the file creation for both normal file uploads and for markdown
+     * @param file The file to be uploaded
+     * @param keepFileName specifies if original file name should be kept
+     * @param markdown boolean which is set to true, when we are uploading a file within the markdown editor
+     * @return The path of the file
+     */
+    @NotNull
+    public String handleSaveFile(MultipartFile file, boolean keepFileName, boolean markdown) {
+        // NOTE: Maximum file size is set in resources/config/application.yml
+        // Currently set to 10 MB
+
+        // check for file type
+        String filename = file.getOriginalFilename();
+        if (filename == null) {
+            throw new IllegalArgumentException("Filename cannot be null");
+        }
+        // sanitize the filename and replace all invalid characters with "_"
+        filename = filename.replaceAll("[^a-zA-Z\\d\\.\\-]", "_");
+        String fileExtension = FilenameUtils.getExtension(filename);
+        if (this.allowedFileExtensions.stream().noneMatch(fileExtension::equalsIgnoreCase)) {
+            throw new BadRequestException("Unsupported file type! Allowed file types: " + String.join(", ", this.allowedFileExtensions));
+        }
+
+        final String filePath;
+        final String fileNameAddition;
+        final StringBuilder responsePath = new StringBuilder();
+
+        // set the appropriate values depending on the use case
+        if (markdown) {
+            filePath = FilePathService.getMarkdownFilePath();
+            fileNameAddition = "Markdown_";
+            responsePath.append("/api/files/markdown/");
+        }
+        else {
+            filePath = FilePathService.getTempFilePath();
+            fileNameAddition = "Temp_";
+            responsePath.append("/api/files/temp/");
```

Comment:
The "/api/files/temp/" is used in multiple places in this file. Would it make sense to extract a constant for it?